packages/hub/routes/profiles.ts

```diff
@@ -0,0 +1,253 @@
+      profile.textColor = this.sanitizeText(ctx.request.body.data.attributes['text-color']);
+    }
+
+    if (attributes['profile-description']) {
+      profile.profileDescription = this.sanitizeText(ctx.request.body.data.attributes['profile-description']);
+    }
+
+    if (attributes['profile-image-url']) {
+      profile.profileImageUrl = this.sanitizeText(ctx.request.body.data.attributes['profile-image-url']);
+    }
+
+    if (attributes['links']) {
+      profile.links = ctx.request.body.data.attributes['links'];
+    }
+
+    let errors = await this.profileValidator.validate(profile);
+    let hasErrors = Object.values(errors).flatMap((i) => i).length > 0;
+
+    if (hasErrors) {
+      ctx.status = 422;
+      ctx.body = {
+        errors: serializeErrors(errors),
+      };
+    } else {
+      await prisma.profile.update({
+        data: {
+          profileDescription: profile.profileDescription,
+          profileImageUrl: profile.profileImageUrl,
+          links: profile.links,
+        },
+        where: { id: profile.id },
+      });
+
+      let serialized = await this.profileSerializer.serialize(profile);
+
+      ctx.status = 200;
+      ctx.body = serialized;
+    }
+    ctx.type = 'application/vnd.api+json';
+  }
+
+  async getValidation(ctx: Koa.Context) {
+    if (!ensureLoggedIn(ctx)) {
+      return;
+    }
+
+    let slug: string = ctx.params.slug;
+    let validationResult = await this.validateSlug(slug);
+
+    ctx.status = 200;
+    ctx.body = validationResult;
+    ctx.type = 'application/vnd.api+json';
+  }
+
+  async validateSlug(slug: string) {
+    let errorMessage = validateMerchantId(slug);
+
+    if (errorMessage) {
+      return {
+        slugAvailable: false,
+        detail: errorMessage,
+      };
+    } else {
+      if (
+        this.reservedWords.isReserved(slug, (reservedWord) => reservedWord.replace(/[^0-9a-zA-Z]/g, '').toLowerCase())
+      ) {
+        return {
+          slugAvailable: false,
+          detail: 'This ID is not allowed',
+        };
+      } else {
+        let prisma = await this.prismaManager.getClient();
+        let profile = await prisma.profile.findFirst({ where: { slug } });
+        return {
+          slugAvailable: profile ? false : true,
+          detail: profile ? 'This ID is already taken. Please choose another one' : 'ID is available',
+        };
+      }
+    }
+  }
+
+  async getFromShortId(ctx: Koa.Context) {
+    let prisma = await this.prismaManager.getClient();
+    let profile = await prisma.profile.findUnique({
+      where: {
+        id: shortUUID().toUUID(ctx.params.id),
+      },
+    });
+
+    if (!profile) {
+      ctx.status = 404;
+      return;
+    }
+
+    ctx.status = 200;
+    ctx.body = this.profileSerializer.serialize(profile);
+    ctx.type = 'application/vnd.api+json';
+  }
+
+  sanitizeText(value: string | unknown): any {
+    if (typeof value === 'string') {
+      return value.trim().replace(/ +/g, ' ');
```

Comment:
More pre-existing problems - Is this method doing what it claims to do? It seems like it's just normalizing spaces at the moment. If the goal is to normalize spaces and we don't care about newlines, we should use `/\s+/g` instead. 

Not sure if we need to fix this elsewhere too

Reply:
I don’t know the intent but I’m reluctant to dig into it really, my goal was to just preserve existing behaviour from the two merged endpoints, but I do think we should look at what we really want here

Reply:
Can we rename this so that we don't forget that it is not really sanitising text atm?